File: Java/Easy/HouseRobber.java
class Solution {
    public int rob(int[] nums) {
    if(nums.length == 0)
      return 0;
    int valueSoFar = 0;
    int currentLoot = 0;
    for(int num: nums){
      int temp = valueSoFar;
      valueSoFar = Math.max(currentLoot + num, valueSoFar);
      currentLoot = temp;
    }
    return valueSoFar;
    }
}

File: Sprint/HouseRobber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
        h1 = h2 = 0
        for num in nums:
            temp = h1
            h1 = max(num + h2, h1)
            h2 = temp
        return h1
